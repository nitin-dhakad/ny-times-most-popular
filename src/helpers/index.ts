/**
 * write a method to get publish day and month
 * @param {*} url function take as input url
 * @param {*} params function n params in order to replace in API
 * @return {*} updated url with provied parameters
 */

export const updateURLParams = (url: string, ...params: string[]) => {
  let updatedUrl = url;
  params.forEach((param, index) => {
    updatedUrl = updatedUrl.replace(`{{param${index + 1}}}`, param);
  });
  return updatedUrl;
};

/**
 * write a method to fetch media image url
 * @param {*} media function take as input
 * @param {*} quality image quality to be fetched expected values 0 - low , 1- good, 2 - best
 * @return {*} media image url
 */
export const extractImageUrl = (
  media: { [x: string]: { [x: string]: string }[] }[],
  quality: number
): string => {
  return media?.[0]?.["media-metadata"]?.[quality]?.["url"] || "";
};

/**
 * write a method to get publish day and month
 * @param {*} date function take as input published date
 * @return {*} day and month
 */
export const getDateAndMonth = (date: string): string => {
  const dateObj = new Date(date);
  const month = dateObj.toLocaleString("default", { month: "long" });
  const day = dateObj.getDate();
  return day + " " + month;
};
