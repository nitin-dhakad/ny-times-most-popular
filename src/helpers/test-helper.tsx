import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { setupStore } from "../store";



export const renderWithProviders = (
  ui: React.ReactElement,
  { route = "/", ...renderOptions } = {}
) => {
  window.history.pushState({}, "Test page", route);
  return render(
    <Provider store={setupStore()}>
      <BrowserRouter>{ui}</BrowserRouter>
    </Provider>,
    renderOptions
  );
};
