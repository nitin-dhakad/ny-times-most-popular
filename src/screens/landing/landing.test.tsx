// import { render } from "@testing-library/react";
import Landing from ".";
import { renderWithProviders } from "../../helpers/test-helper";

describe("rendering landing page", () => {
  test("show landing page", () => {
    const { container } = renderWithProviders(<Landing />);
    expect(container).toMatchSnapshot();
  });
});

