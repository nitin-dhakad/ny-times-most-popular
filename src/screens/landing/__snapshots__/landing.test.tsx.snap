// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering landing page show landing page 1`] = `
<div>
  <div
    class=" justify-center items-center flex-col"
  >
    <div
      class="flex flex-row items-center justify-between  p-4"
    >
      <h1
        class="text-4xl font-semibold font-serif italic text-center flex-1 ml-52  text-slate-800"
      >
        The New York Times
      </h1>
      <div
        class="flex justify-end mr-1"
      >
        <div
          class="text-base font-bold text-gray-600 pr-1"
        >
          Select Day(s):
          <select
            class="text-base font-bold text-gray-600 ml-4 px-3 text-center border-solid border-2 border-gray-800 rounded"
          >
            <option
              class="text-base text-gray-500 text-left"
              value="1"
            >
              1 day
            </option>
            <option
              class="text-base text-gray-500 text-left"
              value="7"
            >
              7 days
            </option>
            <option
              class="text-base text-gray-500 text-left"
              value="30"
            >
              30 days
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      class="fixed inset-0 z-50 flex justify-center items-center "
    >
      <div
        aria-busy="true"
        aria-label="triangle-loading"
        class="sc-braxZu juNRvt undefined"
        data-testid="triangle-loading"
        role="progressbar"
      >
        <svg
          class="sc-elDIKY jSkGDu"
          data-testid="triangle-svg"
          height="80"
          id="triangle"
          viewBox="-3 -4 39 39"
          width="80"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            class="sc-gJhJTp fgpNzY"
            fill="transparent"
            points="16,0 32,32 0,32"
            stroke="#4fa94d"
            stroke-width="1"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
